### PR TITLE
test: Added tests, which check whether App/API initializers are overriding

### PR DESCRIPTION
**Pull request by mikeyusko**
# Summary of Changes

Added tests for `falcon.API`, `falcon.App` initializers, which check whether a `media_type` is overriding.

# Related Issues

https://github.com/falconry/falcon/issues/1577

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](https://github.com/falconry/falcon/blob/master/CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] Added **tests** for changed code.
- [ ] Prefixed code comments with GitHub nick and an appropriate prefix.
- [x] Coding style is consistent with the rest of the framework.
- [ ] Updated **documentation** for changed code.
    - [ ] Added docstrings for any new classes, functions, or modules.
    - [ ] Updated docstrings for any modifications to existing code.
    - [ ] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [ ] Updated all relevant supporting documentation files under `docs/`.
    - [ ] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [ ] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [ ] Changes (and possible deprecations) have [towncrier](https://pypi.org/project/towncrier/) news fragments under `docs/_newsfragments/`. (Run `towncrier --draft` to ensure it renders correctly.)

If you have *any* questions to *any* of the points above, just **submit and ask**! This checklist is here to *help* you, not to deter you from contributing!

*PR template inspired by the attrs project.*

### CODE REVIEW

File: tests/test_app_initializers.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import falcon
+import falcon.testing as testing
```
Comment (vytas7) [inline tests/test_app_initializers.py]: A cosmetic nitpick, but this `import` form looks rather unorthodox; normally one uses `import as` to rename imported names. Maybe just `from falcon import testing`?

File: tests/test_app_initializers.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class MediaTypeResource:
+    def on_get(self, req, resp):
+        resp.body = 'foo bar'
+
+
+def test_api_media_type_overriding():
```
Comment (vytas7) [inline tests/test_app_initializers.py]: Could we DRY these two test cases a bit? (Potentially with a little help from `pytest.mark.parametrize(...)`)

Comment (mikeyusko) [reply tests/test_app_initializers.py]: @vytas7 Fixed 👍

File: tests/test_app_initializers.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    handlers = falcon.media.Handlers({
+        'text/html': MediaHTMLHandler()
+    })
+    app.resp_options.media_handlers = handlers
+
+    return testing.TestClient(app)
+
+
+@pytest.mark.parametrize('client', (falcon.App,), indirect=True)
```
Comment (kgriffs) [inline tests/test_app_initializers.py]: Could this be merged with `test_api_media_type_overriding()` by adding `falcon.API` to the _argvalues_ ? E.g.:

```py
@pytest.mark.parametrize('client', (falcon.App, falcon.API), indirect=True)
```

Comment (mikeyusko) [reply tests/test_app_initializers.py]: Yes, will do.

**Pull request merged**